src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -287,6 +290,10 @@ protected string GenerateCommandLineCommands (string ManifestFile, string curren
 			} else if (!string.IsNullOrEmpty (VersionCodePattern)) {
 				manifest.CalculateVersionCode (null, VersionCodePattern, VersionCodeProperties);
 			}
+			if (!manifest.ValidateVersionCode (out string error)) {
+				Log.LogCodedError ("APT0000", ManifestFile, 0, error);
```

Comment:
An invalid `versionCode` value should not be an `APT0000` error, so that we can distinguish these failures in telemetry.  We should introduce a new XA.... code, presumably XA0002? https://github.com/xamarin/xamarin-android/tree/master/Documentation/guides/messages